static/app/views/dashboards/widgets/tableWidget/utils.ts

```diff
@@ -13,3 +19,29 @@ export function convertTableDataToTabularData(tableData?: TableData): TabularDat
     },
   };
 }
+
+/**
+ * Takes columns and field aliases list, converting them to a record for `TableWidgetVisualization`.
+ * If field header map is supplied, it will combine them, giving priority to field aliases on
+ * duplicate keys
+ *
+ * @param columns a `TabularColumn[]`. Length and order must match `fieldAliases`
+ * @param fieldAliases a list of strings. Length and order must match `columns`
+ * @param fieldHeaderMap optional `Record<string, string>` to combine `fieldAliases` with
+ */
+export function decodeColumnAliases(
+  columns: TabularColumn[],
+  fieldAliases: string[],
+  fieldHeaderMap?: Record<string, string>
```

Comment:
```suggestion
  fieldHeaderMap: Record<string, string> = {}
```
I think this is a little bit cleaner, you can avoid checking for `undefined` below. Not a big deal though

Reply:
good suggestion!!

Reply:
ok cursor hates it LOL 😭